modules/RTCStats/RTCStats.ts

```
@@ -0,0 +1,189 @@
+     * The proxies will then send data to the rtcstats server via the trace object.
+     * The initialization procedure must be called once when lib-jitsi-meet is loaded.
+     * 
+     * @param initConfig initial config for rtcstats.
+     * @returns {void}
+     */
+    init(initConfig: any) {        
+        const {
+            analytics: {
+                rtcstatsUseLegacy: useLegacy = false,
+                rtcstatsPollInterval: pollInterval= 10000,
+                rtcstatsSendSdp: sendSdp = false,
+                rtcstatsEnabled = false
+            } = {}
+        } = initConfig;
+
+        // If rtcstats is not enabled or already initialized, do nothing.
+        // Calling rtcsatsInit multiple times will cause the global objects to be rewritten multiple times,
+        // with unforseseen consequences.
+        if (!rtcstatsEnabled || this._initialized) return;
+        
+        rtcstatsInit( 
+            { statsEntry: this.sendStatsEntry.bind(this) }, 
+            { connectionFilter,
+              pollInterval,
+              useLegacy,
+              sendSdp,
+              eventCallback: (event) => this.events.emit(RTC_STATS_PC_EVENT, event)}
+        );
+
+        this._initialized = true;
+    }
+
+    /**
+     * When a conference is about to start, we need to reset the trace module, and initialize it with the
+     * new conference's config. On a normal conference flow this wouldn't be necessary, as the whole page is
+     * reloaded, but in the case of breakout rooms or react native the js context doesn't reload, hence the
+     * RTCStats singleton and its config persists between conferneces.
+     * 
+     * @param conference - JitsiConference instance that's about to start.
+     * @returns {void}
+     */
+    start(conference: JitsiConference) {
+        // If rtcstats proxy module is not initialized, do nothing.
+        if (!this._initialized) return;
```

When would this happen?

---

Ideally it wouldn't, it's just meant as some basic invariant checking. But if you prefer a fail fast approach I can remove it.

---

Let's add a log line at least, since it implies something is bad.

---

Added.